core/lib/Cardano/Address/Script.hs

```diff
@@ -452,6 +453,10 @@ validateScriptTemplate
     -> ScriptTemplate
     -> Either ErrValidateScriptTemplate ()
 validateScriptTemplate level (ScriptTemplate cosigners' script) = do
+    first WrongScript $ do
+        requiredValidation script
+        when (level == RecommendedValidation ) $
+            first NotRecommended (recommendedValidation script)
```

Comment:
It can use the new function, right?

```suggestion
    first WrongScript $ validateScriptInScriptTemplate level (ScriptTemplate cosigners' script)
```